### DISCUSSION TITLE
person identification

### DISCUSSION THREAD

**Discussion by balajeek** [photoprism/photoprism#1467] [Q&A] [answered]
Just setup yesterday and started the index of my 10,000+ photos/videos library.  its still indexing for last 20+ hrs.

Questions:
- Is it going to finish indexing at some point and after that i will only index new files or change?
- Is it possible i can name a person in the photo and it can tag other photos.  so if i search for a name of a person in the family it will bring all photos.

**Comment by TheUnityGuy** [comment, marked as answer]
Yes, it is going to finish at some point, I can even say that it's going to happen in near future if you use a decent CPU and good hard drive. For example on my raspberry pi 4 (1.5GHz arm CPU and slow af HDD) it took about 48 hours to finish indexing ~7000 photos. You can also check already indexed photos quantity in Library/Originals section of the main UI.
Every time you add a new file to originals folder, Photoprism will index only that file. You can also reindex the whole library if you want.
![Screenshot_2021-08-11-17-56-08-95_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://user-images.githubusercontent.com/71348891/129064745-dec7f84c-9dbe-4902-9d00-f4f0bc820dbe.jpg)


Unfortunately no. Photoprism can't recognize  people's faces yet. There's a roadmap card  on GitHub that you can investigate for more information. I guess in a few months (or years 😅) this feature will come to Photoprism.

**Reply by balajeek** [reply]
thanks for the info, i do see the progress under Library tab, my qnap is intel celero 1.5 Ghz 16gb mem, so far good.
about facaial regonization, I first noticed Photonix in reddit where they mentioned this and was going to install that, then found they don't index video files, then ended up with photoprism.

**Reply by TheUnityGuy** [reply]
Yeah, I also previously used Photonix and PiGallery2 but they are incomparable to Photoprism even in this 'beta' state. It is well developed and community is also big so I guess there will be a lot of new features in next releases :))

**Reply by balajeek** [reply]
good to know, thank you.

**Reply by graciousgrey** [reply]
We are currently working on face recognition. So it should hopefully be finished rather in weeks/a month than years ;)

**Reply by balajeek** [reply]
wow! thats great news, can't wait for it.
All these photos on my nas sitting without looked the program like prims helps family members easily scrubb thru and have fun moments.  Thanks.

**Reply by graciousgrey** [reply]
Facial recognition available now, see: https://docs.photoprism.org/user-guide/organize/people/
